static/js/settings_data.js

```diff
@@ -176,6 +176,16 @@ export function user_can_edit_topic_of_any_message() {
     return user_has_permission(page_params.realm_edit_topic_policy);
 }
 
+export function user_can_delete_own_message() {
+    if (
+        page_params.realm_delete_own_message_policy ===
+        settings_config.common_message_policy_values.by_everyone.code
+    ) {
+        return true;
+    }
+    return user_has_permission(page_params.realm_delete_own_message_policy);
```

Comment:
Sam comment that `user_has_permission` should have the code for the `everyone` common code, rather than it being in an individual function like this.